src/app.rs

```diff
@@ -176,76 +152,86 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
                 .unwrap_or_default(),
         );
 
-        let app_init_cfg = AppInitCfg {
-            mount_type: MountType::Takeover,
-            phantom: PhantomData,
-            into_after_mount: Box::new({
-                let base_path = Rc::clone(&base_path);
-                move |url: Url,
-                      orders: &mut OrdersContainer<Ms, Mdl, INodes, GMs>|
-                      -> AfterMount<Mdl> {
-                    let url = url.skip_base_path(&base_path);
-                    let model = init(url, orders);
-                    AfterMount::new(model).url_handling(UrlHandling::None)
-                }
-            }) as Box<dyn IntoAfterMount<Ms, Mdl, INodes, GMs>>,
+        let app = Self {
+            cfg: Rc::new(cfg::AppCfg::new(
+                util::window().document().expect("get window's document"),
+                // @TODO_B: trim out to remove mount_point field
+                root_element.get_element().expect("get root element"),
+                update,
+                view,
+                base_path,
+            )),
+            data: Rc::new(AppData {
+                model: RefCell::new(None),
+                main_el_vdom: RefCell::new(None),
+                popstate_closure: RefCell::new(None),
+                hashchange_closure: RefCell::new(None),
+                window_event_handler_manager: RefCell::new(EventHandlerManager::new()),
+                // @TODO_B: We are trying to get rid of using SubManager
+                sub_manager: RefCell::new(crate::app::sub_manager::SubManager::new()),
+                msg_listeners: RefCell::new(Vec::new()),
+                scheduled_render_handle: RefCell::new(None),
+                after_next_render_callbacks: RefCell::new(Vec::new()),
+                render_info: Cell::new(None),
+            }),
         };
-        let app = Self::new(
```

Comment:
Removing this abstraction made it easier reason about the changes I needed to make. Most likely trivial to revert if preferred.